code/modules/mining/machine_silo.dm

```diff
@@ -8,6 +11,9 @@
 	interaction_flags_machine = INTERACT_MACHINE_WIRES_IF_OPEN|INTERACT_MACHINE_ALLOW_SILICON|INTERACT_MACHINE_OPEN_SILICON
 	processing_flags = NONE
 
+	/// Currently selected UI tab, possible values are: `MACHINE_VIEW` or `1` and `LOGS_VIEW` or `0`.
+	var/current_view = MACHINE_VIEW
```

Comment:
Dont do this. You are looking at this from your own localhost perspective where 1 player opens 1 UI and not from a multiplayer server perspective.

What happens if 2 players open 2 UI's and 1 player opens the machine tab but the 2nd player opens the log tab? Player A will receive a blank page or worst an TGUI blue screen because `current_view` was set by the last player that changed tabs.

Keep the original code, send all data and let the client side UI decide which data to view

Reply:
Oh, that makes sense, thank you for the review.